### ISSUE TITLE
Allow string array as an argument for AuthGuard and defaultStrategy option.

### ISSUE THREAD

**Issue by michaeljota** [nestjs/passport#67]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Though Passport allow for an array of string to be passed as an argument for the declared strategies, AuthGuard, and defaultStrategy does not. 

## Expected behavior

From nestjs/nest#2116:

Be able to specifify multiple authtentication middlewares | authGuards to protect the same resources, let's say for example if a haver this controller but want to be able to authenticate for twitter and google.
In this case the controller method should be accesible whether im logged in via twitter or google

## Minimal reproduction of the problem with instructions

```ts
@UseGuards(AuthGuard(['google', 'twitter']))
                     ~~~~~~~~~~~~~~~~~~~~~ // <-- Argument of type 'string[]' is not assignable to parameter of type 'string'. 
async sayHello(){
  return  "hello"
}
```

## What is the motivation / use case for changing the behavior?

This module should allow to interact with Passport as if we were interacting with Passport directly.

## Environment

<pre><code>
Nest Passport version: 6.0.0

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Done #68. Published as 6.1.0

**Comment by vegerot** [comment]
What about having multiple `AuthGuard`s?  Such as `@UseGuards([AuthGuard('foo'), BarAuthGuard])`?
